src/org/zaproxy/zap/extension/ascanrulesAlpha/IntegerOverflow.java

```diff
@@ -0,0 +1,271 @@
+			setParameter(msg, param, returnAttack);
+			sendAndReceive(msg);
+			requestReturn = msg.getResponseHeader();			
+			// This is where BASE baseResponseBody was you detect potential vulnerabilities in the response
+    		chkerrorheader = requestReturn.getHeadersAsString();
+    		log.debug("Header: "+ chkerrorheader);
+    		if (msg.getResponseHeader().getStatusCode() == HttpStatusCode.INTERNAL_SERVER_ERROR)
+    		{
+    			log.debug("Found Header");
+    			bingo(getRisk(), 
+    					Alert.CONFIDENCE_MEDIUM, 
+    					this.getBaseMsg().getRequestHeader().getURI().toString(), 
+    					param, 
+    					msg.getRequestHeader().toString(), 
+    					this.getError('2') ,
+    					msg);
+    			return;
+    		}
+    		msg = getNewMsg();
+			returnAttack = singleString(4,'1');// The number of full length ints to send.
+			setParameter(msg, param, returnAttack);
+			sendAndReceive(msg);
+			requestReturn = msg.getResponseHeader();			
+			// This is where BASE baseResponseBody was you detect potential vulnerabilities in the response
+    		chkerrorheader = requestReturn.getHeadersAsString();
+    		log.debug("Header: "+ chkerrorheader);
+    		if (msg.getResponseHeader().getStatusCode() == HttpStatusCode.INTERNAL_SERVER_ERROR)
+    		{
+    			log.debug("Found Header");
+    			bingo(getRisk(), 
+    					Alert.CONFIDENCE_MEDIUM, 
+    					this.getBaseMsg().getRequestHeader().getURI().toString(), 
+    					param, 
+    					msg.getRequestHeader().toString(), 
+    					this.getError('3') ,
+    					msg);
+    			return;
+    		}
+    		msg = getNewMsg();
+			returnAttack = singleString(4,'9');// The number of full length ints to send.
+			setParameter(msg, param, returnAttack);
+			sendAndReceive(msg);
+			requestReturn = msg.getResponseHeader();			
+			// This is where BASE baseResponseBody was you detect potential vulnerabilities in the response
+    		chkerrorheader = requestReturn.getHeadersAsString();
+    		log.debug("Header: "+ chkerrorheader);
+    		if (msg.getResponseHeader().getStatusCode() == HttpStatusCode.INTERNAL_SERVER_ERROR)
+    		{
+    			log.debug("Found Header");
+    			bingo(getRisk(), 
+    					Alert.CONFIDENCE_MEDIUM, 
+    					this.getBaseMsg().getRequestHeader().getURI().toString(), 
+    					param, 
+    					msg.getRequestHeader().toString(), 
+    					this.getError('4') ,
+    					msg);
+    			return;
+    		}
+    			
+				return;	
+
+			
+		} catch (Exception e) {
+			log.error(e.getMessage(), e);
+		}	
+	}
+
+	@Override
+	public int getRisk() {
+		return Alert.RISK_MEDIUM;
+	}
+
+	@Override
+	public int getCweId() {
+		// The CWE id
+		return 190;
+	}
+
+	@Override
+	public int getWascId() {
+		// The WASC ID
+		return 3;
+	}
+	
+	private String randomIntegerString(int length)
+	{
+		
+		int numbercounter = 0;
+		int character = 0;
+		long charactercounter = 0;
+		long maxcharacter = 11;
+		StringBuilder sb1 = new StringBuilder();

```

Comment:
The initial capacity of the `StringBuilder` could be set to `length * maxcharacter` (if `maxcharacter` is an `int`, or add the cast). The same comment to `StringBuilder` of `singleString` method.